scalapb-runtime/src/main/scala/scalapb/ByteStringDecoder.scala

```diff
@@ -0,0 +1,14 @@
+package scalapb
+
+import com.google.protobuf.ByteString
+
+@FunctionalInterface
+trait ByteStringDecoder[T] {
+  def apply(bs: ByteString): T
+}
+
+case object ByteStringDecoder {
```

Comment:
What's the intent for using `case object` here and not `object`? The code base uses `object` so for consistency I would prefer that here as well as in other location, unless of course there's a technical reason.